File: 이것이 취업을 위한 코딩 테스트다/구현/상하좌우/상하좌우.java
package 구현.상하좌우;

import java.util.*;

public class 상하좌우 {
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);
    int size = sc.nextInt();
    sc.nextLine();
    String str = sc.nextLine();

    int currentRow = 1;
    int currentColumn = 1;
    int dx[] = {-1, 1, 0, 0};
    int dy[] = {0, 0, -1, 1};
    char dt[] = {'U', 'D', 'L', 'R'};
    int nextRow, nextColumn;

    for (int i = 0; i < str.length(); i+=2) {
      char type = str.charAt(i);

      for (int j = 0; j < 4; j++) {
        if (type == dt[j]) {
          nextRow = currentRow + dx[j];
          nextColumn = currentColumn + dy[j];

          if (1 <= nextRow && nextRow <= size && 1 <= nextColumn && nextColumn <= size) {
            currentRow = nextRow;
            currentColumn = nextColumn;
          }
        }

      }
    }

    System.out.println(currentRow + " " + currentColumn);
  }
}

File: 이것이 취업을 위한 코딩 테스트다/구현/상하좌우/상하좌우.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
  int size;
  string str;

  cin >> size;
  cin.ignore();
  getline(cin, str);

  int row[] = {-1, 1, 0, 0};
  int column[] = {0, 0, -1, 1};
  int currentRow = 1, currentColumn = 1;
  char type[] =  {'U', 'D', 'L', 'R'};

  for(int i = 0; i < str.size(); i+=2) {
    char dType = str[i];

    for (int j = 0; j < 4; j++) {
      if (dType == type[j]) {
        int nextRow = currentRow + row[j];
        int nextColumn = currentColumn + column[j];

        if (1 <= nextRow && nextRow <= size && 1 <= nextColumn && nextColumn <= size) {
          currentRow = nextRow;
          currentColumn = nextColumn;
        }
      }
    }
  }

  cout << currentRow << ' ' << currentColumn;

  return 0;
}